src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4509,11 +4509,31 @@ private void VisitDeconstructionArguments(ArrayBuilder<DeconstructionVariable> v
                     }
                     else
                     {
-                        // https://github.com/dotnet/roslyn/issues/33006: Check nullability of `this` argument.
-                    }
+                        // Check nullability for `this` parameter
```

Comment:
Is there a way to re-use the implementation of `VisitArguments` (line 2669) instead of duplicating some/most of it?

Reply:
Not easily. Its a path I started down previously, but ended up being pretty messy too. Happy to give it another go though, if we think it'd be better factored.

---
In reply to: [277465231](https://github.com/dotnet/roslyn/pull/35016#discussion_r277465231) [](ancestors = 277465231)